controller/control.go

```diff
@@ -234,6 +234,11 @@ func (c *Controller) Start(addresses ...string) error {
 
 func (c *Controller) WriteAt(b []byte, off int64) (int, error) {
 	c.RLock()
+	if off < 0 || off+int64(len(b)) > c.size {
```

Comment:
`./scripts/integration-test -- -k test_beyond_bounary`(found I wrote a typo here)

After `make build`, you need to refresh you local binary for this to run correctly.